src/vs/workbench/api/common/extHostTypeConverters.ts

```diff
@@ -3335,23 +3335,42 @@ export namespace LanguageModelToolResult2 {
 			if (item.kind === 'text') {
 				return new types.LanguageModelTextPart(item.value);
 			} else if (item.kind === 'data') {
-				const mimeType = Object.values(types.ChatImageMimeType).includes(item.value.mimeType as types.ChatImageMimeType) ? item.value.mimeType as types.ChatImageMimeType : undefined;
-				if (!mimeType) {
-					throw new Error('Invalid MIME type');
-				}
-				return new types.LanguageModelDataPart(item.value.data.buffer, mimeType);
+				// TODO: make sure we can remove this
+				// const mimeType = Object.values(types.ChatImageMimeType).includes(item.value.mimeType as types.ChatImageMimeType) ? item.value.mimeType as types.ChatImageMimeType : undefined;
+				// // if (!mimeType) {
+				// // 	throw new Error('Invalid MIME type');
+				// // }
+				return new types.LanguageModelDataPart(item.value.data.buffer, item.value.mimeType);
```

Comment:
We would need to make sure the data type is handled well in the chat extension. I suggest thinking about alternative representations for the content in the chat API. Models are limited to varying degress with what they can represent, it's quite likely an extension may want to show a nice visualization for data that should otherwise be given to the model as text.

We also need to make sure that the user can see exactly what the model sees in some way to protect against unknowning prompt injection attacks.

Reply:
Ah good find. I think I will actually revert this part. It was left over from a different direction I tried